src/auth/reference_modules/saml.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/python3
+import io
+import json
+import os
+
+from onelogin.saml2.auth import OneLogin_Saml2_Auth
+from onelogin.saml2.settings import OneLogin_Saml2_Settings
+
+ENTRA_IDP_ROLE_ATTRIBUTE = "http://schemas.microsoft.com/ws/2008/06/identity/claims/role"  # NOSONAR
+# ^ standardized by Microsoft: https://learn.microsoft.com/en-us/entra/identity-platform/reference-saml-tokens
+
+# Default settings: "sp" refers to the service provider (Memgraph), and "idp" to the identity provider (Entra ID)
+SETTINGS_TEMPLATE = {
+    "strict": True,
+    "debug": True,
+    "sp": {
+        "entityId": "",
+        "assertionConsumerService": {
+            "url": "https://memgraph.com/<acs>",
+            # Dummy value required by library (this service is Memgraph-internal, not online)
+        },
+        "NameIDFormat": "urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified",
+        "x509cert": "",
+        "privateKey": "",
+    },
+    "idp": {
+        "entityId": "",
+        "singleSignOnService": {
+            "url": "https://app.onelogin.com/trust/saml2/http-post/sso/<onelogin_connector_id>",
+            # Dummy value required by library (the library would use it to send auth requests,
+            # whereas this module only does auth)
+        },
+        "x509cert": "",
+    },
+    "security": {
+        "wantAssertionsEncrypted": False,
+        "wantNameIdEncrypted": False,
+    },
+}
+
+
+# Dummy request (python3-saml library assumes there’s one, and you can only pass the response together with it)
+def mock_request(scheme_env):
+    return {
+        "http_host": "localhost:3000",
+        "script_name": os.environ.get(f"AUTH_SAML_{scheme_env}_CALLBACK_URL", ""),
+    }
+
+
+def load_from_file(path: str):
+    if not path:
+        return ""
+
+    with open(path, mode="r") as file:
+        return file.read()
+
+
+def str_to_bool(string: str):
+    return string.lower() == "true"
+
+
+def has_role_attribute(scheme: str, attributes):
+    if scheme == "saml-entra-id":
+        return ENTRA_IDP_ROLE_ATTRIBUTE in attributes.keys()
+    elif scheme == "saml-okta":
+        return os.environ.get("AUTH_SAML_OKTA_ROLE_ATTRIBUTE", "") in attributes.keys()
+    return False
+
+
+def get_username(auth, attributes, use_nameid, username_attribute):
+    if use_nameid:
+        return auth.get_nameid()
+
+    if username_attribute not in attributes.keys():
+        raise Exception("Username attribute not supplied.")
+
+    if isinstance(attributes[username_attribute], list):
+        return attributes[username_attribute][0]
+
+    return attributes[username_attribute]
+
+
+def get_role_mappings(raw_role_mappings: str):
+    if not raw_role_mappings:
+        return {}
+
+    role_mappings = "".join(raw_role_mappings.split(" "))  # remove whitespace
+    return {mapping.split(":")[0]: mapping.split(":")[1] for mapping in role_mappings.split(";")}
```

Comment:
I think it's maybe better to have a path to json file that contains role mappings since it can be big

Reply:
Considered this; think we’re fine for now:
* the current way is consistent with the `--auth-module-mappings` flag, the only other SSO-related mapping